firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -345,28 +358,30 @@ private Void registerAndSaveFid(PersistedInstallationEntry persistedInstallation
   }
 
   /** Calls the FIS servers to generate an auth token for this Firebase installation. */
-  private InstallationTokenResult fetchAuthTokenFromServer(
-      PersistedInstallationEntry persistedInstallationEntry) throws FirebaseInstallationsException {
+  private void fetchAuthTokenFromServer(PersistedInstallationEntry persistedInstallationEntry)
+      throws FirebaseInstallationsException {
     try {
       long creationTime = utils.currentTimeInSecs();
-      InstallationTokenResult tokenResult =
+      TokenResult tokenResult =
           serviceClient.generateAuthToken(
               /*apiKey= */ firebaseApp.getOptions().getApiKey(),
               /*fid= */ persistedInstallationEntry.getFirebaseInstallationId(),
               /*projectID= */ firebaseApp.getOptions().getProjectId(),
               /*refreshToken= */ persistedInstallationEntry.getRefreshToken());
 
-      persistedInstallation.insertOrUpdatePersistedInstallationEntry(
-          PersistedInstallationEntry.builder()
-              .setFirebaseInstallationId(persistedInstallationEntry.getFirebaseInstallationId())
-              .setRegistrationStatus(RegistrationStatus.REGISTERED)
-              .setAuthToken(tokenResult.getToken())
-              .setRefreshToken(persistedInstallationEntry.getRefreshToken())
-              .setExpiresInSecs(tokenResult.getTokenExpirationTimestamp())
-              .setTokenCreationEpochInSecs(creationTime)
-              .build());
-
-      return tokenResult;
+      if (tokenResult.isSuccessful()) {
+        persistedInstallation.insertOrUpdatePersistedInstallationEntry(
+            persistedInstallationEntry
+                .toBuilder()
+                .setRegistrationStatus(RegistrationStatus.REGISTERED)
+                .setAuthToken(tokenResult.getToken())
+                .setExpiresInSecs(tokenResult.getTokenExpirationTimestamp())
+                .setTokenCreationEpochInSecs(creationTime)
+                .build());
+      } else if (tokenResult.isErrored()) {
+        persistedInstallation.clear();
+      }
```

Comment:
add a comment saying what's the *else" after this, and why you do a non-op for it.